packages/kit/src/core/sync/create_manifest_data/index.spec.js

```diff
@@ -507,36 +507,31 @@ test('creates routes with named layouts', () => {
 	]);
 
 	assert.equal(routes.filter((route) => route.page).map(simplify_route), [
-		{
-			id: 'a/a1',
-			pattern: '/^/a/a1/?$/',
-			page: { layouts: [0, 4], errors: [1, undefined], leaf: 10 }
-		},
 		{
 			id: '(special)/a/a2',
 			pattern: '/^/a/a2/?$/',
 			page: { layouts: [0, 2], errors: [1, undefined], leaf: 9 }
 		},
 		{
-			id: 'b/c/c2',
-			pattern: '/^/b/c/c2/?$/',
-			page: { layouts: [0], errors: [1], leaf: 11 }
+			id: 'a/a1',
+			pattern: '/^/a/a1/?$/',
+			page: { layouts: [0, 4], errors: [1, undefined], leaf: 10 }
```

Comment:
Is having this one _after_ `(special)/a/a2` correct? I suppose in this case it isn't going to matter, but do we need to worry about how we're handling URLs with params in them? Is `(foo)/[b=b]` going to come before `[a=a]` because of the `(`? Ordinarily `[a=a]` would come before `[b=b]`, and it seems weird to me if that changes because of a group that doesn't affect any of the URLs.

Reply:
`[a=a]` and `[b=b]` are sorted lexically, and so are `(foo)/[b=b]` and `[a=a]`. I get what you're saying but this feels simpler to me than 'identical-for-the-purposes-of-ranking routes are sorted lexically but only after ignoring their groups'. I don't know that there's an objectively right answer here — I think that if you have this level of ambiguity in your routes then you're making a rod for your own back regardless of what we do

Reply:
Yup, that's fair.